vms/platformvm/service.go

```diff
@@ -511,6 +511,83 @@ func (s *Service) GetUTXOs(_ *http.Request, args *api.GetUTXOsArgs, response *ap
 	return nil
 }
 
+/*
+ *****************************************************
+ ******************* Get Subnet **********************
+ *****************************************************
+ */
+
+// GetSubnetArgs are the arguments to GetSubnet
+type GetSubnetArgs struct {
+	// ID of the subnet to retrieve information about
+	SubnetID ids.ID `json:"subnetID"`
+}
+
+// GetSubnetResponse is the response from calling GetSubnet
+type GetSubnetResponse struct {
+	// ID of the subnet
+	SubnetID ids.ID `json:"subnetID"`
+	// whether it is permissioned or not
+	IsPermissioned bool `json:"isPermissioned"`
+	// subnet auth information for a permissioned subnet
+	ControlKeys []string    `json:"controlKeys"`
+	Threshold   json.Uint32 `json:"threshold"`
+	// subnet transformation tx ID for a permissionless subnet
+	SubnetTransformationTxID ids.ID `json:"subnetTransformationTxID"`
+}
+
+func (s *Service) GetSubnet(_ *http.Request, args *GetSubnetArgs, response *GetSubnetResponse) error {
+	s.vm.ctx.Log.Debug("API called",
+		zap.String("service", "platform"),
+		zap.String("method", "getSubnet"),
+	)
+
+	s.vm.ctx.Lock.Lock()
+	defer s.vm.ctx.Lock.Unlock()
+
+	response.SubnetID = args.SubnetID
+
+	if response.SubnetID == constants.PrimaryNetworkID {
+		response.IsPermissioned = false
+		response.ControlKeys = []string{}
+		response.Threshold = json.Uint32(0)
+		response.SubnetTransformationTxID = ids.Empty
+		return nil
+	}
+
+	if subnetTransformationTx, err := s.vm.state.GetSubnetTransformation(response.SubnetID); err == nil {
```

Comment:
We should check for the actual error type - so that we don't report an unexpected error as incorrect information.